app/v1/search.py

```diff
@@ -0,0 +1,65 @@
+import json
+import threading
+from typing import Any, Dict, cast
+from matplotlib.pyplot import title
+
+import numpy as np
+import pandas as pd
+import structlog
+from fastapi import APIRouter
+from enum import Enum
+
+from app import utils
+
+from .schemas import (
+    Dimension,
+    DimensionProperties,
+    VariableMetadataResponse,
+    VariableSource,
+)
+
+log = structlog.get_logger()
+
+
+router = APIRouter()
+
+
+class SearchType(str, Enum):
+    table = "meta_tables"
+    variable = "meta_variables"
+    dataset = "meta_datasets"
+
+
+@router.get("/search")
+def search(term: str, type: SearchType = SearchType.variable, limit: int = 10):
+    con = utils.get_readonly_connection(threading.get_ident())
+
+    # TODO: implement search on other tables too? not sure whether we'll need it yet
+    if type != SearchType.variable:
+        raise NotImplementedError()
```

Comment:
nit: add a message? `NotImplementedError("Invalid search type {type}, only searching variables is currently supported")`